### PR TITLE
feat [IDP-1984] :  Added the initial docs for Custom Plugins

### PR DESCRIPTION
**Pull request by Debanitrkl**
# Harness Developer Pull Request
Added the initial docs for Custom Plugins

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/internal-developer-portal/plugins/custom-plugins/overview.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+sidebar_label: Overview
+description: Learn about custom plugins in Internal Developer Portal and how to use them to customize IDP.
+sidebar_position: 1
+---
+
+The Custom Plugins feature in IDP will allow users to plug their own private Backstage plugins into the Harness IDP and use them along with the curated plugins available for everyone. The development and ownership of these plugins is dependent on the users. 
+
+:::info
+
+At present we only support the **Frontend Plugins**
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/plugins/custom-plugins/overview.md:12]: We should also explain why that is the case. 

We allow you to create your own UI components, and use the Backstage proxy and Delegates, to connect to your internal systems and show meaningful data on the IDP UI.

**Review by OrkoHunter — COMMENTED** [review]
We should launch this along with the release of custom plugins behind a feature flag. cc @sandeepa-kv-jois @jenil-harness

File: docs/internal-developer-portal/plugins/custom-plugins/add-a-custom-plugin.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+- **Card**: Here's an [example](https://github.com/harness/backstage-plugins/blob/cd70f9c03ed52917d8f409c2ceb7b7fc874c97de/packages/app/src/components/catalog/EntityPage.tsx#L192) `EntityLinksCard` of Links card on **Overview** page.
+
+![](./static/links-card.png)
+
+- **Tab**: Here's an [example](https://github.com/harness/backstage-plugins/blob/cd70f9c03ed52917d8f409c2ceb7b7fc874c97de/packages/app/src/components/catalog/EntityPage.tsx#L87) `EntityHarnessCiCdContent` of Harness CI/CD tab. 
+
+![](./static/ci-cd-tab.png)
+
+- **Menu Item in SideNav**: Here's an example of TODO plugin in side nav.
```
Comment (jenil-harness) [inline docs/internal-developer-portal/plugins/custom-plugins/add-a-custom-plugin.md:129]: Conditional as well.

File: docs/internal-developer-portal/plugins/custom-plugins/add-a-custom-plugin.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+9. If you require `app-config.yaml` for your plugin, then **enable configuration**. 
+
+10. Now add the proxy part here, as given below in the example,
+
+```YAML
+proxy:
+  /mynewplugin:
+    target: https://somerandomapi.com/api/v2
+    pathRewrite:
+      /api/proxy/dynatrace/?: /
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/plugins/custom-plugins/add-a-custom-plugin.md:148]: This doesn't look right

File: docs/internal-developer-portal/plugins/custom-plugins/add-a-custom-plugin.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
+---
+title: Add custom plugins in IDP
+description: Learn about the steps to add your custom plugin in IDP.
+sidebar_position: 2
+---
+
+:::info
+
+This Feature is in **BETA** phase and is behind a Feature Flag. It's only enabled on few accounts on request.
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/plugins/custom-plugins/add-a-custom-plugin.md:9]:
```suggestion
This Feature is in **BETA** phase and is behind <name_of_flag> Feature Flag. Please contact the IDP product team if you would like to try out Custom Plugins.
```

File: docs/internal-developer-portal/plugins/custom-plugins/add-a-custom-plugin.md (hunk 3 above)
Comment (Debanitrkl) [reply docs/internal-developer-portal/plugins/custom-plugins/add-a-custom-plugin.md:148]: Aah missed it will remove dynatrace from there

**Review by jenil-harness — APPROVED** [review]
LGTM

**Pull request merged**